File: mobile/lib/models/permission.dart
class Permission {
  final bool aiAnswerEnabled;
  final bool aiExplanationEnabled;
  final bool aiSimilarQuestionsEnabled;

  Permission({
    required this.aiAnswerEnabled,
    required this.aiExplanationEnabled,
    required this.aiSimilarQuestionsEnabled,
  });

  factory Permission.fromJson(Map<String, dynamic> json) => Permission(
        aiAnswerEnabled: json['ai_answer_enabled'] ?? true,
        aiExplanationEnabled: json['ai_explanation_enabled'] ?? true,
        aiSimilarQuestionsEnabled: json['ai_similar_questions_enabled'] ?? true,
      );

  Map<String, dynamic> toJson() => {
        'ai_answer_enabled': aiAnswerEnabled,
        'ai_explanation_enabled': aiExplanationEnabled,
        'ai_similar_questions_enabled': aiSimilarQuestionsEnabled,
      };
}

File: server/app/models/permission.py
from datetime import datetime, timezone
from sqlalchemy import Integer, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Permission(Base):
    __tablename__ = "permissions"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("users.id"), unique=True, index=True)

    ai_answer_enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    ai_explanation_enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    ai_similar_questions_enabled: Mapped[bool] = mapped_column(Boolean, default=True)

    change_log: Mapped[list | None] = mapped_column(JSON, nullable=True)

    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )

    user = relationship("User", back_populates="permissions")
